### DISCUSSION TITLE
setup a Samba share with network discovery using docker-compose

### DISCUSSION THREAD

**Discussion by giscus** [FarisZR/FarisZR.com#4] [Comments]
# setup a Samba share with network discovery using docker-compose

Easily setup an SMB share with network discovery for MacOS and windows using docker and Docker-compose

https://fariszr.com/en/docker-smb-network-discovery/

**Comment by toondaey** [comment]
Thanks for this article! 
So I followed the instructions to the letter but can't seem to get the discovery working. I also decided to create another container in order to access the smb container from within but so far I've only been able to get authentication working but can't access the shared directory. I'm using an Mac with M1 pro chip.
Please assist.

**Reply by FarisZR** [reply]
The WSDD container doesn't support ARM, but that shouldn't affect macOS discovery.
Maybe there is some sort of quirk with docker desktop on macOS, as AFAIK it creates a complete VM for Docker to run, so are you sure that the container actually can see the files?

**Reply by toondaey** [reply]
I think it has something to do with [this](https://docs.docker.com/network/host/)
```
The host networking driver only works on Linux hosts, and is not supported on Docker Desktop for Mac, Docker Desktop for Windows, or Docker EE for Windows Server.
```

**Comment by Retrograde-i486** [comment]
Thank you for this excellent article, I had to setup a samba share for my new HP printer (HP Color LaserJet MFP M283cdw if anyone stumbles upon this while googling) as it lacks a scan-to-ftp setting like my old Brother printer had and this worked perfectly!

**Comment by paulk1997** [comment]
So I am having an issue with access.  I can see the share in Window.  I can browse to it without issue but the folder is empty and I can't create a folder.  The user I have specified is the owner of the directory and all files.

## https://fariszr.com/en/docker-smb-network-discovery/
version: "3.5"

services:
  samba:
    image: crazymax/samba
    container_name: samba
    network_mode: host
    volumes:
      - ./data:/data
      - /plex:/samba/plex:rw
      - /shared:/samba/shared:rw
    environment:
      - "america/chicago"
      - "SAMBA_LOG_LEVEL=0"
      - "WSDD2_ENABLE=1"
      - "WSDD2_NETBIOS_NAME=docker_samba"
    restart: always

config.yml
auth:
  - user: paulk1997
    group: paulk1997
    uid: 1000
    gid: 1000
    password: password

global:
  - "force user = paulk1997"
  - "force group = paulk1997"

share:
  - name: shared
    comment: shared
    path: /samba/shared
    browsable: yes
    readonly: no
    guestok: yes
    veto: no
    recycle: yes
  - name: plex
    path: /samba/plex
    browsable: yes
    readonly: no
    guestok: yes
    veto: no


The 1000 user owns all file.

**Reply by FarisZR** [reply]
hmmm, maybe the user 1000 is used inside the container already and has a different name, try to test that by opening a shell inside it.
i faced the same issue as well.
